lib/db/lowlevel.go

```diff
@@ -256,9 +256,24 @@ func (b *batch) checkFlush() {
 }
 
 func (b *batch) flush() {
-	if err := b.db.Write(b.Batch, nil); err != nil && err != leveldb.ErrClosed {
+	tran, err := b.db.OpenTransaction()
```

Comment:
Should this return an error instead of a panic?

Reply:
Yes and no. It would definitely be nice if we actually tracked errors, but the current state is that we either panic or ignore errors all over the db package. Returning an error here would just mean panicking or ignoring one level up. Changing this would require a whole lot more changes.